custom/ewsghana/alerts/alerts.py

```diff
@@ -0,0 +1,408 @@
+from celery.schedules import crontab
+from celery.task import periodic_task
+import datetime
+from casexml.apps.stock.models import StockTransaction
+from corehq.apps.commtrack.models import SupplyPointCase, StockState
+from corehq.apps.locations.models import Location, SQLLocation
+from corehq.apps.products.models import SQLProduct
+from corehq.apps.sms.api import send_sms_to_verified_number
+from corehq.apps.users.models import CommCareUser
+from custom.ewsghana.alerts import ONGOING_NON_REPORTING, ONGOING_STOCKOUT_AT_SDP, ONGOING_STOCKOUT_AT_RMS,\
+    REPORT_REMINDER, DOMAIN, WEB_REMINDER, URGENT_NON_REPORTING, URGENT_STOCKOUT, COMPLETE_REPORT, INCOMPLETE_REPORT, \
+    BELOW_REORDER_LEVELS, ABOVE_THRESHOLD, WITHOUT_RECEIPTS
+from django.core.mail import send_mail
+import settings
+from corehq.apps.commtrack.models import CommtrackConfig
+
+
+def send_alert(transactions, sp, user, message):
+    sp_ids = set()
+    if sp and not transactions and user.get_verified_number():
+        send_sms_to_verified_number(user.get_verified_number(), message)
+        sp_ids.add(sp._id)
+
+
+# Alert when facilities have not been reported continuously for 3 weeks
+@periodic_task(run_every=crontab(hour=10, minute=00),
+              queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def on_going_non_reporting():
+    sp_ids = set()
+    now = datetime.datetime.utcnow()
+    date = now - datetime.timedelta(days=21)
+
+    for user in CommCareUser.by_domain(DOMAIN):
```

Comment:
Handler should work for all domains which have EWS integration enabled. Please look at this function: https://github.com/dimagi/commcare-hq/blob/master/custom/ewsghana/reminders/reminders.py#L32